pkg/i2gw/providers/common/converter.go

```diff
@@ -365,9 +397,17 @@ func (rg *ingressRuleGroup) configureBackendRef(servicePorts map[types.Namespace
 			continue
 		}
 		backendRefs = append(backendRefs, gatewayv1.HTTPBackendRef{BackendRef: *backendRef})
+		
+		// Track source for this backend
+		sources = append(sources, intermediate.BackendSource{
+			Ingress: path.sourceIngress,
+			Path:    &path.path,
+		})
 	}
 
-	return removeBackendRefsDuplicates(backendRefs), errors
+	// Remove duplicates while maintaining source tracking
+	// uniqueBackendRefs, uniqueSources := removeBackendRefsDuplicatesWithSources(backendRefs, sources)
```

Comment:
hmmm, what to do if we have duplicates with different sources?

Reply:
in the IR maybe, but the resulted backendRef, in the final route should not be the same.

Reply:
I think this dedup came out of a bug where we had two backednRefs that are exactly the same resulted in a final route output